cmd/grype/cli/cli.go

```diff
@@ -70,6 +72,14 @@ func create(id clio.Identification) (clio.Application, *cobra.Command) {
 		).
 		WithPostRuns(func(_ *clio.State, _ error) {
 			stereoscope.Cleanup()
+		}).
+		WithMapExitCode(func(err error) int {
+			// return exit code 2 to indicate when a vulnerability severity is discovered
+			// that is equal or above the given --fail-on severity value.
+			if errors.Is(err, grypeerr.ErrAboveSeverityThreshold) {
+				return 2
```

Comment:
I think this makes the most sense: error code 1 is "something bad happened, we don't know what it is" and an error code > 1 is a report of a specific condition. We also have [code 100 for db update checks](https://github.com/anchore/grype/blob/0ca1080d18f23fb21c89073651af316e532d39c5/cmd/grype/cli/commands/db_check.go#L88). I think we could handle that condition here, too, if we add a specific error type.

I wonder if this would end up breaking many users or there is a high probability that users are just checking for the nonzero return. I do think the way you've updated here makes the most sense to me. I'll ask the team's opinion on this detail.

Reply:
I wholeheartedly hope nobody checks exit=1, best way to shoot yourself in the foot 😄

Good idea to move the check for `exitCodeOnDBUpgradeAvailable` here. When I was going through the code base for alternative exit code, I noticed it but forgot about it. Also the fact that os.Exit is called here means no defers in the callpath will run, which may or may not matter for this particular command, but should probably still be addressed.

I'll add a commit to do just that.

Reply:
Yeah, it would be great to get rid of that `os.Exit` call 👍